aleph-client/src/contract/mod.rs

```diff
@@ -168,8 +177,8 @@ impl ContractInstance {
             self.address.clone(),
             value,
             Weight {
-                ref_time: Self::MAX_GAS,
-                proof_size: Self::MAX_GAS,
+                ref_time: self.max_gas_override.unwrap_or(DEFAULT_MAX_GAS),
+                proof_size: self.max_gas_override.unwrap_or(DEFAULT_MAX_GAS),
```

Comment:
These are almost never equivalent and usually much different. Maybe there should be a way to override them separately in the final version of the client.

Reply:
What does it mean _in our case_? When I dry run any contract (bulletin board, psp22, etc.) these two are always different and non-zero.

Reply:
if i'm not mistaken, it is non-zero, because that's what pallets' dispatchables return, but it isn't incurred

below you have both weights non-zero, but proof_size is not included in the actual fee (there's the standard weight + some length fee)
![image](https://user-images.githubusercontent.com/27450471/214005814-5e08ff4c-8a09-4ae4-a6fd-0a9f160dddb1.png)

Reply:
The thing is you have to provide "enough" `proofSize` limit to get the txn executed - that's how it works with the JS lib (i.e. I can't get txn to execute if `proofSize == 0`). If we are not charging the `proofSize` cost then it's even worse IMO b/c it's confusing: we require people to estimate/set it properly but in the end we will not charge for that? Even though it is returned in the final result?

Reply:
added possibility to override proof size limit